app/models/task.py

```diff
@@ -1,6 +1,32 @@
 from flask import current_app
 from app import db
 
-
+# child class 
 class Task(db.Model):
-    task_id = db.Column(db.Integer, primary_key=True)
+    task_id = db.Column(db.Integer, primary_key=True, autoincrement=True)
+    title = db.Column(db.String)
+    description = db.Column(db.String)
+    completed_at = db.Column(db.DateTime, nullable = True, default = None)
+    goal_id = db.Column(db.Integer, db.ForeignKey('goal.goal_id'), nullable = True)
+    goal=db.relationship("Goal",backref=db.backref('task',lazy=True))
+
+    
+    # helper function that will find out if a task has been completed or not
+    def is_completed(self):
+        return self.completed_at != None      
+    # helper function to output for response body 
+    def to_json(self):
+        return {
+            "id": self.task_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_completed()
+        }   
+    def goal_json(self):
+        return {
+            "id": self.task_id,
+            "goal_id": self.goal_id,
+            "title": self.title,
+            "description": self.description,
+            "is_complete": self.is_completed()
+        }       
```

Comment:
Good work building this instance methods as helpers. Consider DRYing up your code by building the dictionary 

```
json = 
{"id": self.task_id,                    
"title": self.title,
 "description": self.description,
"is_complete": self.check_if_complete()}
```
and conditionally adding the `goal_id` like this:

```
if self.goal_id:
    json["goal_id] = self.goal_id
```

and then returning `{"task": json}`